tools/runners/run-mongo.js

```diff
@@ -607,38 +607,43 @@ var launchMongo = function (options) {
         });
       }
 
-      var initiateResult = yieldingMethod(
-        db.admin(), 'command', {replSetInitiate: configuration});
+      try{
+        var initiateResult = yieldingMethod(
+          db.admin(), 'command', {replSetInitiate: configuration});
+      } catch (e){
+        if(e.message !== 'already initialized') {
+          throw Error("rs.initiate error: " + e.message);
+        }
+      }
+
       if (stopped) {
         return;
       }
-      // why this isn't in the error is unclear.
-      if (initiateResult && initiateResult.documents
-          && initiateResult.documents[0]
-          && initiateResult.documents[0].errmsg) {
-        var err = initiateResult.documents[0].errmsg;
-        if (err !== "already initialized") {
-          throw Error("rs.initiate error: " +
-                      initiateResult.documents[0].errmsg);
-        }
-      }
+
       // XXX timeout eventually?
       while (!stopped) {
-        var status = yieldingMethod(db.admin(), 'command',
-                                    {replSetGetStatus: 1});
-        if (!(status && status.documents && status.documents[0])) {
-          throw status;
+        var status;
+        try{
+          status = yieldingMethod(db.admin(), 'command',
+                                      {replSetGetStatus: 1});
+        } catch (e) {
+          // Some of the expected results, come as errors.
+          status = e;
         }
-        status = status.documents[0];
+
+        // See https://docs.mongodb.com/manual/reference/replica-states/
+        // for information on various states
         if (!status.ok) {
-          if (status.startupStatus === 6) {  // "SOON"
+          // UNKNOWN(6) some of the other replicas are in an UNKNOWN state from
+          // this replica perspective.
+          if (status.myState === 6) {
```

Comment:
I'm not sure we should include this code unless we can reproduce the code path. 

It looks like the original test was included in this commit: https://github.com/Fabs/meteor/commit/1d5da6b4390512857f222f3e8959dc5dda505093 used by this commit: https://github.com/Fabs/meteor/commit/72c9657b0e8a7b209ded1df67f22dd80e2fab0a2

You can run that test with `./meteor self-test 'mongo failover'`. Perhaps you should try running that and see if we can nail this down.